pkg/ddc/thin/engine.go

```diff
@@ -114,13 +114,23 @@ func Precheck(client client.Client, key types.NamespacedName) (found bool, err e
 }
 
 // CheckReferenceDatasetRuntime judge if this runtime is used for handling dataset mounting another dataset.
-func CheckReferenceDatasetRuntime(client client.Client, runtime *datav1alpha1.ThinRuntime) (bool, error) {
-	dataset, err := utils.GetDataset(client, runtime.Name, runtime.Namespace)
-	if err != nil {
+func CheckReferenceDatasetRuntime(ctx cruntime.ReconcileRequestContext, runtime *datav1alpha1.ThinRuntime) (bool, error) {
+	dataset, err := utils.GetDataset(ctx.Client, runtime.Name, runtime.Namespace)
+	if err != nil && utils.IgnoreNotFound(err) != nil {
+		// ignore dataset not found err and try to get mounted dataset from runtime
```

Comment:
I think the comment should be added below? This is the case which does not ignore error.